python-avd/pyavd/_cv/workflows/models.py

```diff
@@ -192,6 +195,55 @@ class DeployToCvResult:
     removed_device_tags: list[CVDeviceTag] = field(default_factory=list)
     removed_interface_tags: list[CVInterfaceTag] = field(default_factory=list)
 
+    def update_from_workflow_devices(self, workflow_devices: list[WorkflowDevice]) -> None:
+        # iterate thru each workflow device and update deployed/skipped/removed placeholder
+        # for device tags/interface tags/config/pathfinder metadata in overall deployment result
+        for device in workflow_devices:
+            # iterate thru all device_tags checking for deployment status and accumulate under correct bucket
+            for device_tag in device.device_tags:
+                match device_tag.status:
+                    case DeploymentStatus.SKIPPED:
+                        self.skipped_device_tags.append(device.get_one_cv_tag(device_tag.input))
+                    case DeploymentStatus.DEPLOYED:
```

Comment:
There is a type checking issue here (and below). Should we explicitly check type of a return to make checker happy?

```suggestion
                    case DeploymentStatus.SKIPPED:
                        if isinstance(cv_tag := device.get_one_cv_tag(device_tag.input), CVDeviceTag):
                            self.skipped_device_tags.append(cv_tag)
                    case DeploymentStatus.DEPLOYED:
```

Reply:
Thank you for your suggestion, made changes.